tests/integration/api/controllers/contact.spec.js

```diff
@@ -0,0 +1,549 @@
+        path: `${endpoint}?${stringQueryParams}`,
+        auth: adminUser
+      };
+      const firstPage = await utils.request(opts);
+
+      // second request
+      queryParams.cursor = firstPage.cursor;
+      stringQueryParams = new URLSearchParams(queryParams).toString();
+      const opts2 = {
+        path: `${endpoint}?${stringQueryParams}`,
+        auth: adminUser
+      };
+      const secondPage = await utils.request(opts2);
+
+      const allData = [...firstPage.data, ...secondPage.data];
+
+      expect(allData).excludingEvery(['_rev', 'reported_date']).to.deep.equalInAnyOrder(expectedContactIds);
+      expect(firstPage.data.length).to.be.equal(2);
+      expect(secondPage.data.length).to.be.equal(1);
+      expect(firstPage.cursor).to.be.equal('2');
+      expect(secondPage.cursor).to.be.equal(null);
+    });
+
+    it('returns a page of place type contact ids when limit and cursor is passed and cursor can be reused',
+      async () => {
+        // first request
+        const expectedContactIds = [place0._id, clinic1._id, clinic2._id];
+        const queryParams = {
+          freetext: placeFreetext,
+          type: placeType,
+          limit: twoLimit
+        };
+        let stringQueryParams = new URLSearchParams(queryParams).toString();
+        const opts = {
+          path: `${endpoint}?${stringQueryParams}`,
+          auth: adminUser
+        };
+        const firstPage = await utils.request(opts);
+
+        // second request
+        queryParams.cursor = firstPage.cursor;
+        stringQueryParams = new URLSearchParams(queryParams).toString();
+        const opts2 = {
+          path: `${endpoint}?${stringQueryParams}`,
+          auth: adminUser
+        };
+        const secondPage = await utils.request(opts2);
+
+        const allData = [...firstPage.data, ...secondPage.data];
+
+        expect(allData).excludingEvery(['_rev', 'reported_date']).to.deep.equalInAnyOrder(expectedContactIds);
+        expect(firstPage.data.length).to.be.equal(2);
+        expect(secondPage.data.length).to.be.equal(1);
+        expect(firstPage.cursor).to.be.equal('2');
+        expect(secondPage.cursor).to.be.equal(null);
+      });
+
+    it(`throws error when user does not have can_view_contacts permission`, async () => {
+      const opts = {
+        path: `/api/v1/contact/id`,
+        auth: { username: userNoPerms.username, password: userNoPerms.password },
+      };
+      await expect(utils.request(opts)).to.be.rejectedWith('403 - {"code":403,"error":"Insufficient privileges"}');
+    });
+
+    it(`throws error when user is not an online user`, async () => {
+      const opts = {
+        path: `/api/v1/contact/id`,
+        auth: { username: offlineUser.username, password: offlineUser.password },
+      };
+      await expect(utils.request(opts)).to.be.rejectedWith('403 - {"code":403,"error":"Insufficient privileges"}');
+    });
+
+    it('throws 400 error when contactType is invalid', async () => {
```

Comment:
suggestion: did I miss it somewhere, or do we also need a test case where freetext is empty and no type is provided?